File: homy/integrations/zabbix/widgets/status/status.js
window.WidgetRegistry.register('zabbix_status', {
    async render(container, widgetData, config) {
        container.innerHTML = '<div class="widget-loading"><div class="spinner"></div></div>';

        try {
            const data = await API.request(`/api/zabbix_status/data?widget_id=${widgetData.id}`);
            container.innerHTML = '';

            if (!data.online) {
                const msg = data.message || window.i18n?.t('zabbix_not_configured', 'integration_zabbix')
                    || 'Zabbix not available';
                container.innerHTML = `<div class="widget-error text-center" style="padding:12px;">${msg}</div>`;
                return;
            }

            const fmt = (n) => (window.formatNumber ? window.formatNumber(n) : Number(n).toLocaleString());
            const high = data.problems_high || 0;
            const statusClass = high > 0 ? 'zabbix-alert' : 'zabbix-ok';

            container.innerHTML = `
                <div class="zabbix-status-card">
                    <div class="zabbix-status-header">
                        <span class="zabbix-brand">
                            <i data-lucide="activity" class="zabbix-icon ${statusClass}"></i>
                            <span>Zabbix</span>
                        </span>
                        <span class="zabbix-version muted-text">v${data.version || '?'}</span>
                    </div>
                    <div class="zabbix-metrics">
                        <div class="zabbix-metric">
                            <span class="metric-label">Hosts</span>
                            <span class="metric-value">${fmt(data.host_count || 0)}</span>
                        </div>
                        <div class="zabbix-metric">
                            <span class="metric-label">Problems</span>
                            <span class="metric-value ${data.problems_count ? 'text-warning' : ''}">${fmt(data.problems_count || 0)}</span>
                        </div>
                        <div class="zabbix-metric">
                            <span class="metric-label">High / Disaster</span>
                            <span class="metric-value ${high ? 'text-danger' : ''}">${fmt(high)}</span>
                        </div>
                    </div>
                    <div class="zabbix-severity-row muted-text">
                        <span>W: ${data.problems_warning || 0}</span>
                        <span>Avg: ${data.problems_average || 0}</span>
                        <span>Disaster: ${data.problems_disaster || 0}</span>
                    </div>
                </div>
            `;
            window.refreshLucideIcons(container);
        } catch (err) {
            container.innerHTML = `<div class="widget-error"><i data-lucide="alert-triangle"></i><span>${err.message}</span></div>`;
            window.refreshLucideIcons(container);
        }
    },
});

File: homy/integrations/zabbix/widgets/status/status.py
"""Zabbix overview widget — hosts, problems, version."""

import logging

from flask import jsonify, request

from homy.cache import cache_key, should_bypass_cache, widget_cache
from homy.integration_widget_util import get_integration_for_widget, get_integration_plugin_module

logger = logging.getLogger(__name__)

WIDGET = {
    'type': 'zabbix_status',
    'name': 'Zabbix Overview',
    'integration_types': ['zabbix'],
    'default_size_x': 6,
    'default_size_y': 4,
    'icon': 'activity',
    'config_schema': {
        'integration_id': {'type': 'text', 'label': 'Integration ID', 'default': ''},
    },
}

CACHE_NS = 'zabbix_status'
CACHE_TTL = 60


def register(app):
    @app.route('/api/zabbix_status/data', methods=['GET'], endpoint='widget_zabbix_status_data')
    def zabbix_status_data():
        widget_id = request.args.get('widget_id')
        if not widget_id:
            return jsonify({'error': 'Bad Request', 'message': 'widget_id required'}), 400

        ck = cache_key(CACHE_NS, widget_id)
        if not should_bypass_cache():
            cached = widget_cache.get(ck)
            if cached is not None:
                return jsonify(cached)

        _, config, err, code = get_integration_for_widget(widget_id, 'zabbix')
        if err:
            return jsonify({'configured': False, 'online': False, 'message': err}), code

        if not config.get('server_url', '').strip() or not config.get('api_token', '').strip():
            return jsonify({
                'configured': False,
                'online': False,
                'message': 'Zabbix URL and API token required',
            })

        try:
            zabbix_mod = get_integration_plugin_module('zabbix')
            summary = zabbix_mod.fetch_summary(config)
            result = {
                'configured': True,
                'online': True,
                'version': summary.get('version', ''),
                'host_count': summary.get('host_count', 0),
                'problems_count': summary.get('problems_count', 0),
                'problems_high': summary.get('problems_high', 0),
                'problems_disaster': summary.get('problems_disaster', 0),
                'problems_warning': summary.get('problems_warning', 0),
                'problems_average': summary.get('problems_average', 0),
                'by_severity': summary.get('by_severity', {}),
            }
            widget_cache.set(ck, result, ttl=CACHE_TTL)
            return jsonify(result)
        except Exception as exc:
            logger.warning('Zabbix status failed: %s', exc)
            return jsonify({
                'configured': True,
                'online': False,
                'message': str(exc),
            }), 200
